x-pack/plugins/fleet/server/index.ts

```diff
@@ -53,6 +53,9 @@ export const config: PluginConfigDescriptor = {
     unused('agents.pollingRequestTimeout', { level: 'critical' }),
     unused('agents.tlsCheckDisabled', { level: 'critical' }),
     unused('agents.fleetServerEnabled', { level: 'critical' }),
+    // Removed default policies
+    unused('agentPolicies.is_default', { level: 'critical' }),
+    unused('agentPolicies.is_default_fleet_server', { level: 'critical' }),
```

Comment:
@joshdover should we keep these unused records if we are keeping the flags now? they are still deprecated

Reply:
Are we still reading these config values during Fleet setup? If so, then I think we should add a warning log that these will be removed, but I don't think we can use the `unused` helper because it will remove the field from the config object that gets passed to Fleet I believe.

I think you'll need to implement a custom deprecation function here rather than use the provided ones to return a non-critical warning.

Reply:
yes, the flags can be used in kibana config.

I found that there is a `deprecate` function which seems to be doing what we want:
```  
   * Deprecate a configuration property from inside a plugin's configuration path.
     * Will log a deprecation warning if the deprecatedKey was found.
     *
     * @example
     * Log a deprecation warning indicating 'myplugin.deprecatedKey' should be removed by `8.0.0`
     * ```typescript
     * const provider: ConfigDeprecationProvider = ({ deprecate }) => [
     *   deprecate('deprecatedKey', '8.0.0'),
     * ]
     * 
  ```

Reply:
updated with deprecation warnings